core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -327,6 +330,8 @@ public Object get(int i) {
       case 15:
         return sortOrderId;
       case 16:
+        return partitionSpecId;
+      case 17:
```

Comment:
It looks like this is a bug from https://github.com/apache/iceberg/pull/1723/files. I think this should return `fileOrdinal` and not `pos`!

Reply:
Yea you are right, I can file issue and take a look at it after this change is in.

Reply:
https://github.com/apache/iceberg/issues/3167